smarts/sstudio/genscenario.py

```diff
@@ -121,13 +133,22 @@ def gen_scenario(
         )
 
 
+def gen_map(scenario: str, map_spec: types.MapSpec, output_dir: str = None):
+    output_path = os.path.join(output_dir or scenario, "map_spec.pkl")
+    with open(output_path, "wb") as f:
+        # we use cloudpickle here instead of pickle because the
+        # map_spec object may contain a reference to a map_builder callable
+        cloudpickle.dump(map_spec, f)
```

Comment:
I think this might be overly unsafe to load a method from a file that could be from an unknown source without any checks.

The first warning `cloudpickle` mentions in documentation is to not load data from un-trusted sources.

> **Security notice**: one should only load pickle data from trusted sources as otherwise pickle.load can lead to arbitrary code execution resulting in a critical security vulnerability.
https://github.com/cloudpipe/cloudpickle#cloudpickle

`cloudpickle` also mentions that `pickle` handles references. Are we doing something dynamic that is causes `pickle` to fail?

> An important difference between cloudpickle and pickle is that cloudpickle can serialize a function or class by value, whereas pickle can only serialize it by reference. Serialization by reference treats functions and classes as attributes of modules, and pickles them through instructions that trigger the import of their module at load time.
https://github.com/cloudpipe/cloudpickle#overriding-pickles-serialization-mechanism-for-importable-constructs

Reply:
As discussed earlier, I've added a comment to warn about this as well as filing Issue #1188 to track this in our backlog.  (Please feel free to update the issue as you see fit.)